src/main/java/com/yahoo/memory/NioBits.java

```diff
@@ -0,0 +1,161 @@
+
+      VM_IS_DIRECT_MEMORY_PAGE_ALIGNED_METHOD =
+          VM_CLASS.getDeclaredMethod("isDirectMemoryPageAligned");
+      VM_IS_DIRECT_MEMORY_PAGE_ALIGNED_METHOD.setAccessible(true);
+      isPageAligned = (boolean)VM_IS_DIRECT_MEMORY_PAGE_ALIGNED_METHOD
+          .invoke(null); //static method
+
+      NIO_BITS_CLASS = Class.forName("java.nio.Bits");
+
+      NIO_BITS_RESERVE_MEMORY_METHOD = NIO_BITS_CLASS
+          .getDeclaredMethod("reserveMemory", long.class, int.class);
+      NIO_BITS_RESERVE_MEMORY_METHOD.setAccessible(true);
+
+      NIO_BITS_UNRESERVE_MEMORY_METHOD = NIO_BITS_CLASS
+          .getDeclaredMethod("unreserveMemory", long.class, int.class);
+      NIO_BITS_UNRESERVE_MEMORY_METHOD.setAccessible(true);
+
+      NIO_BITS_COUNT_FIELD = NIO_BITS_CLASS.getDeclaredField("count");
+      NIO_BITS_COUNT_FIELD.setAccessible(true);
+
+      NIO_BITS_RESERVED_MEMORY_FIELD = NIO_BITS_CLASS.getDeclaredField("reservedMemory");
+      NIO_BITS_RESERVED_MEMORY_FIELD.setAccessible(true);
+
+      NIO_BITS_TOTAL_CAPACITY_FIELD = NIO_BITS_CLASS.getDeclaredField("totalCapacity");
+      NIO_BITS_TOTAL_CAPACITY_FIELD.setAccessible(true);
+
+    } catch (final Exception e) {
+      throw new RuntimeException("Could not acquire java.nio.Bits class: " + e.getClass()
+      + UnsafeUtil.tryIllegalAccessPermit);
+    }
+  }
+
+  static long getCount() {
+    try {
+      final long count = ((AtomicLong)(NIO_BITS_COUNT_FIELD.get(null))).get();
+      return count;
+    } catch (final Exception e) {
+      throw new RuntimeException("Cannot read Bits.count " + e);
+    }
+  }
+
+  static long getReservedMemory() {
+    try {
+      final long resMem = ((AtomicLong)(NIO_BITS_RESERVED_MEMORY_FIELD.get(null))).get();
+      return resMem;
+    } catch (final Exception e) {
+      throw new RuntimeException("Cannot read Bits.reservedMemory " + e);
+    }
+  }
+
+  static long getTotalCapacity() {
+    try {
+      final long resMem = ((AtomicLong)(NIO_BITS_TOTAL_CAPACITY_FIELD.get(null))).get();
+      return resMem;
+    } catch (final Exception e) {
+      throw new RuntimeException("Cannot read Bits.totalCapacity " + e);
+    }
+  }
+
+  static int pageSize() {
+    if (pageSize == -1) {
+      pageSize = unsafe.pageSize();
+    }
+    return pageSize;
+  }
+
+  static int pageCount(final long bytes) {
+    return (int)((bytes + pageSize()) - 1L) / pageSize();
+  }
+
+  static long getMaxDirectByteBufferMemory() {
+    return maxDBBMemory;
+  }
+
+  static boolean isPageAligned() {
+    return isPageAligned;
+  }
+
+  //RESERVE & UNRESERVE BITS MEMORY TRACKING COUNTERS
+  static void reserveMemory(long capacity) {
+    final long pageAdj = isPageAligned ? pageSize : 0;
+    try {
+      //if pageAligned, and if capacity > 2GB, this inflates the reserved size by 1 page per 2GB
+      // or about 2ppm.
+      while (capacity > Integer.MAX_VALUE) {
+        final int chunk = (int) Math.min(capacity, Integer.MAX_VALUE); //(2GB-1) chunks
```

Comment:
I would make 1GB chunks, it feels "safer" to me

Reply:
That is how I had it originally as I had the same thought.  But then it is 2X iterations. I will change it back.